### PR TITLE
Get-AzVm filter by Name prior to checking for throttling

### PR DESCRIPTION
**Pull request by Sandido**
Work for https://msazure.visualstudio.com/One/_workitems/edit/8555684 
and improvement to a previous test method's synopsis. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/VirtualMachine/Operation/GetAzureVMCommand.cs [hunk 1]
```diff
@@ -95,6 +100,37 @@ public class GetAzureVMCommand : VirtualMachineBaseCmdlet
...
+                return resourceManagerClient = resourceManagerClient ?? 
+                    new ResourceManagementClient(
+                        DefaultProfile.DefaultContext.Environment.
+                        GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager), 
+                            ClientCredentials);
+            }
+            set => resourceManagerClient = value;
+        }
+        //test
+
```
Comment (grizzlytheodore) [inline src/Compute/Compute/VirtualMachine/Operation/GetAzureVMCommand.cs:133]: are these used? resourceManagerClinet, ResourceManagerClient, clientCredentials, ClientCredentials

Comment (Sandido) [reply src/Compute/Compute/VirtualMachine/Operation/GetAzureVMCommand.cs]: Removed them.

**Comment by grizzlytheodore** [comment]
LGTM!

**Review by msJinLei — APPROVED** [review]

**Pull request merged**